TypeCobol/Compiler/Symbols/FunctionSymbol.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using TypeCobol.Compiler.Types;
+
+namespace TypeCobol.Compiler.Symbols
+{
+    /// <summary>
+    /// The Symbol of a Function declaration
+    /// </summary>
+    public class FunctionSymbol : ProgramSymbol
```

Comment:
This is convenience mostly for lookup algorithms, and finally a function is generated as a Program.

Reply:
Is it ok to revert the inheritance ? I mean we could say a Program is a Function because all members of a Function are also in a Program (Params, ReturnVar, some data sections). Program add nested programs, functions and  other data sections.

Reply:
Programatically this is true, but I find it counterintuitive from a Cobol perspective to have Program inherit from Function. I would rather go with @smedilol 's suggestion to have a common class for both, also this would give more freedom to have Function evolve without impacting Program.

Reply:
Ok here is my take on this : 
![ClassDiagram1](https://user-images.githubusercontent.com/51322634/82231688-cc657280-992d-11ea-8f45-737bc0c131ad.png)
- `ProgramType` and `FunctionType` are merged into `ScopeType`, this class holds the parameters and return-variable of a `ScopeSymbol`
- `ScopeSymbol` is now an abstraction for both `ProgramSymbol` and `FunctionSymbol`
  - Note that the VisbilityMask properties are `protected abstract`
  - It has a variable collection for all the variables it defines (hence the Scope wording)
  - It has Types, Linkage, Local, Working, Sections, and Paragraphs domains
- `ProgramSymbol` inherits from `ScopeSymbol`
  - It implements the VisibilityMask properties accordingly
  - It also adds Functions, Programs, Global and File domains
- `FunctionSymbol` inherits from `ScopeSymbol`
  - It implements the VisibilityMask properties accordingly

Reply:
Ok for me

Reply:
Ok, my first point of view was oriented code generation as a function will be generated has a cobol program. But as it not our concern here,why not.

Reply:
Refactor done, ProgramSymbol and FunctionSymbol are now in separate branches.